src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -533,10 +533,10 @@ BookStep<TIn, TOut, TDerived>::forEachOffer (
                 // Issuer requires authorization.  See if offer owner has that.
                 auto const& ownerID = offer.owner();
                 auto const authFlag =
-                    ownerID > issuerID ? lsfHighAuth : lsfLowAuth;
+                    ownerID > issuerID ? lsfLowAuth : lsfHighAuth;
 
                 auto const line = afView.read (keylet::line (
-                    ownerID, issuerID, offer.issueOut().currency));
+                    ownerID, issuerID, offer.issueIn().currency));
```

Comment:
Yes, it can be. However, the code still works correctly. The `issuer` from line 530 will be null, and the `if` will prevent this code from running. Even if it did run, it would still work correctly. The trust line for XRP wouldn't exist and the auth flag would not be checked.

Although it may be worth checking for XRP on line 527 as that check should be quick.